ansible/postgresql/init.yml

```diff
@@ -0,0 +1,279 @@
+        login_password: "{{ pg_login_password }}"
+        db: "qms"
+        name: "insert_username"
+      when: install_exts is defined and install_exts and qms is defined and qms
+
+    - name: init db  | create unleash database
+      community.postgresql.postgresql_db:
+        login_host: "{{ pg_login_host }}"
+        port: "{{ pg_listen_port }}"
+        login_user: "{{ pg_login_user }}"
+        login_password: "{{ pg_login_password }}"
+        name: "unleash"
+        owner: "{{ _dbms_connection_user }}"
+        encoding: "{{ encoding }}"
+        lc_collate: "{{ lc_collate }}"
+        lc_ctype: "{{ lc_ctype }}"
+        template: "{{ template }}"
+      when: create_dbs is defined and create_dbs and unleash is defined and unleash
+
+    - name: init db | create staging directory
+      ansible.builtin.tempfile:
+        state: directory
+        suffix: "-dbs"
+      register: dbs_dir
+
+    # - name: init db | download migrate tarball
+    #   ansible.builtin.get_url:
+    #     url: "https://github.com/golang-migrate/migrate/releases/download/{{ migrate_release_version }}/{{ migrate_tarball_name }}"
+    #     dest: "{{ dbs_dir.path }}/{{ migrate_tarball_name }}"
+
+    - name: init db | extract migrate tarball
+      ansible.builtin.unarchive:
+        src: "https://github.com/golang-migrate/migrate/releases/download/{{ migrate_release_version }}/{{ migrate_tarball_name }}"
+        dest: "{{ dbs_dir.path }}"
+        remote_src: yes
+        creates: "{{ dbs_dir.path }}/migrate"
+
+    - name: init db | install migrate
+      become: true
+      become_user: root
+      ansible.builtin.copy:
+        src: "{{ dbs_dir.path }}/migrate"
+        dest: "{{ install_dir }}/migrate"
+        mode: 0755
+
+    - name: init db | checkout de database repo
+      ansible.builtin.git:
+        repo: "https://github.com/cyverse-de/de-database"
+        dest: "{{ dbs_dir.path }}}}/de-database"
+        clone: yes
+        update: yes
+        force: yes
+        version: "{{ de_db_version }}"
+
+    - name: init db | checkout notifications database repo
+      ansible.builtin.git:
+        repo: "https://github.com/cyverse-de/notifications-db"
+        dest: "{{ dbs_dir.path }}/notifications-db"
+        clone: yes
+        update: yes
+        force: yes
+        version: "{{ notifications_db_version }}"
+
+    - name: init db | checkout metadata database repo
+      ansible.builtin.git:
+        repo: "https://github.com/cyverse-de/metadata-db"
+        dest: "{{ dbs_dir.path }}/metadata-db"
+        clone: yes
+        update: yes
+        force: yes
+        version: "{{ metadata_db_version }}"
+
+    - name: init db | run de database migrations
+      ansible.builtin.shell: |
+        migrate -database "postgresql://{{_dbms_connection_user }}:{{_dbms_connection_pass}}@{{ pg_login_host }}:{{ pg_listen_port }}/de?sslmode=disable" -path "/opt/dbs/{{ ansible_user_id}}/de-database/migrations" {{ de_db_migration_cmd }}
```

Comment:
It might be handy to have a `pg_base_url` var or something similar so that these lines could begin with something like `migrate -database {{pg_base_url}}/de?sslmode=disable`

It might also be helpful to URL encode the password just in case it contains characters that need to be escaped. I think you can do that like this: `{{_dbms_connection_pass | urlencode}}`